src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,48 @@
+@import "../utils/variables";
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: $calendar-size;
+  padding: 10px;
+  gap: $gap-size;
+  font-size: 30px;
+  font-family: Arial, Helvetica, sans-serif;
+  cursor: pointer;
+
+  .calendar__day-box {
+    background-color: #eee;
+    width: $box-size;
+    height: $box-size;
+    border: 1px solid black;
+    box-sizing: border-box;
+    text-align: center;
+    line-height: $box-size;
+    transition: transform 0.5s, background-color pink;
+
+    @for $day from 1 through 31 {
+      &--#{$day}::before {
+        content: "#{$day}";
+      }
+    }
```

Comment:
i think this code do nothing here, try to think how to make it work 😉 
you deleted modificator with day, so `--#{$day}` do nothing here, try to use `:nth-child` instead